packages/cli/src/util/is-connected-to-internet.ts

```diff
@@ -0,0 +1,13 @@
+import { resolve as dnsResolve } from 'dns';
+
+export async function isConnectedToInternet() {
```

Comment:
can we memoize this result so that it does not try to repeatedly resolve www.google.com after the initial call?

Reply:
Yeah I can do it, but it just takes a few milliseconds, I think it might be good to request when we need it ( I mean maybe we sometimes cache the connected status but after a few seconds later user gets offline, ...)
WDYT?

Reply:
part of the problem with ahving it sometimes return true or false (ltes say if the connection is flakey) is that these internal checks that are happening if internet is available, but maybe they intermittently dont happen because the internet is down. so like the warning that you are using a local package might not appear, and this would be bad.

i do realize its probably not even a few milliseconds. Just thinking that in general we check to see if internet is available at the beginning--if it isnt then its like we are in a offline mode

Reply:
@dbeal-eth  Yeah you are right, makes sense. I changed it and now it caches the connected to internet status